src/Discord/Parts/Channel/Channel.php

```diff
@@ -350,6 +355,25 @@ public function setPermissions(Part $part, array $allow = [], array $deny = [],
         return $this->setOverwrite($part, $overwrite, $reason);
     }
 
+
+    /**
+     * Rename channel
+     *
+     * @param string $name
+     * @return ExtendedPromiseInterface
+     */
+    public function rename(string $name): ExtendedPromiseInterface
+    {
+        if ($this->name === $name) {
+            return reject(new InvalidOverwriteException('The name is already set in the channel.'));
+        }
+
+        $this->name = $name;
+        return $this->http->patch(Endpoint::bind(Endpoint::CHANNEL, $this->id),[
+            'name' => $this->name
+        ]);
+    }
+
```

Comment:
This should be in a separate PR

Reply:
I disagree with this, there is already `$channel->name` setter and `save()`
we cant let adding every function for changing every simple attribute like this

people would make multiple different http calls just to update a channel information